pkg/scheduler/framework/plugins/clusteraffinity/state.go

```diff
@@ -6,43 +6,84 @@ Licensed under the MIT license.
 package clusteraffinity
 
 import (
-	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework"
 )
 
-type pluginState struct {
-	// requiredAffinityTerms is a list of processed version of required cluster affinity terms.
-	requiredAffinityTerms AffinityTerms
+type observedMinMaxValues struct {
+	min *resource.Quantity
+	max *resource.Quantity
+}
 
-	// preferredAffinityTerms is a list of processed version of preferred cluster affinity terms.
-	preferredAffinityTerms PreferredAffinityTerms
+type pluginState struct {
+	minMaxValuesByProperty map[string]observedMinMaxValues
 }
 
-// preparePluginState initializes the state for the plugin to use in the scheduling cycle.
-// TODO will call this func in the PreFilter stage.
-func preparePluginState(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (*pluginState, error) {
-	if policy.Spec.Policy == nil ||
+func preparePluginState(state framework.CycleStatePluginReadWriter, policy *placementv1beta1.ClusterSchedulingPolicySnapshot) (*pluginState, error) {
+	ps := &pluginState{
+		minMaxValuesByProperty: make(map[string]observedMinMaxValues),
+	}
+
+	// Verify if the policy features any property that requires sorting.
+	noPreferredClusterAffinityTerms := (policy == nil ||
```

Comment:
why do we need this again? It's checked in the preScore

Reply:
Just an extra line of check to avoid implicit assumptions; I do not have any objection removing it though.